tests/system/test_retry.py

```diff
@@ -45,10 +45,10 @@ def test_retry_bubble(echo):
 
     @pytest.mark.asyncio
     async def test_retry_bubble_async(async_echo):
-        with pytest.raises(exceptions.DeadlineExceeded):
+        with pytest.raises(exceptions.RetryError):
```

Comment:
Does this mean that the exception the user sees in case of an error changes? If so, we should flag this as a breaking change, since users may be catching the previously raised exception.

Reply:
It's more like this test case was defined incorrectly. Both `exceptions.DeadlineExceeded` and `exceptoins.RetryError` are raised but we're checking for latter because that confirms that async retry is working as expected.

Reply:
Note that `exceptions.RetryError` was not raised before.